devito/finite_differences/differentiable.py

```diff
@@ -423,14 +423,30 @@ def __new__(cls, *args, **kwargs):
         # a*1 -> a
         args = [i for i in args if i != 1]
 
+        # a*-1 -> a
         # a*-1*-1 -> a
+        # a*-1*-1*-1 -> a*-1
```

Comment:
-a ?

Reply:
the arguments are `a` and `-1` (unless `a` it's an integer) that's why it's not factored out